### ISSUE TITLE
Make CameraCalibrator choose calibration based on Camera ID, not file type

### ISSUE THREAD

**Issue by kosack** [cta-observatory/ctapipe#888]
Right now, CameraCalibrator constructs the calibrator based on the type of the file (as read by the EventSource).  In the future we will have the raw data of multiple cameras in the same format, so we should really choose the calibration type from the camera ID (like "NectarCam") which you can get from the SubarrayDescription that is loaded from an EventSource currently. 

This may require more of a design change, since right now we "calibrate" Monte-Carlo differently than real data (which really is not a good way to work, since in the end they must be identical)

**Comment by kosack** [comment]
TO be more clear, it should be something like this:

```py3
#This:
calib = CameraCalibrator(subarray_description=event.inst.subarray)

# rather than this:
calib = CameraCalibrator(event_source=source)
``

**Comment by kosack** [comment]
Internally it may set up multiple calibrators, covering all telescope types (it could use `subarray.telescope_types` (which is something like `['SST-ASTRI:ASTRICam', 'LST:LSTCam', 'MST:FlashCam']` to decide which ones to use.  The SubarrayDescription also allows mapping of tel_id to type, via `ids = sub.get_tel_ids_for_type("SST-ASTRI:ASTRICam")`

**Comment by dneise** [comment]
Misclick

**Comment by MaxNoe** [comment]
I think this got fixed in #1228
